### DISCUSSION TITLE
feature request: support BCP 47-style language tags

### DISCUSSION THREAD

**Discussion by gao-sun** [jlarmstrongiv/astro-i18n-aut#60] [Ideas]
First, thank you for building this integration!

It seems that currently only 2-letter language tags are supported, such as `fr`. However, [BCP 47](https://www.rfc-editor.org/info/bcp47)-style language tags, such as `zh-Hant`, are not supported by utilities like `getLocale`. It will be great if the integration can support these language tags, as it would allow the integration to reach a broader audience.

**Comment by jlarmstrongiv** [comment]
Thank you for the feature request @gao-sun I’ve converted it into a discussion to talk more about it

I need to do more research into packages that support ISO language codes, ISO script codes, and BCP 47 tags.  That would allow `getLocale` to remain relatively simple without depending on vite virtual modules.

I didn’t initially support tags like `es-ES` and `es-MX` or `en-US` and `en-GB` or BCP 47 because I would only be translating one version myself.  In addition, it’s [best practice](https://stackoverflow.com/a/38959322) to use the shortest code, if available.

For your [use-case](https://stackoverflow.com/a/4894634), are you trying to support both simplified and traditional Chinese?

**Reply by michaelgiraldo** [reply]
Yeah, the goal is to support multiple scripts in the same language - Simplified versus traditional.

Another benefit is that the same language can be used very differently, depending on the country for example the Portugal versus Brazil Portuguese.

**Reply by jlarmstrongiv** [reply]
Sure, I think it’s a worthwhile addition.  I’ve added it to the roadmap in https://github.com/jlarmstrongiv/astro-i18n-aut/discussions/18

Unfortunately, looking at my schedule realistically with holidays and trips coming up, I won’t have a chance to get to it until March.
